Add locale config option

tachiniererin opened the pull request:
This makes the locale of the database created configurable. E.g. for consistent testing where some things depend on the database locale, this is helpful. Or also on broken systems, where the default locale is an invalid string.

I also bumped the versions to the latest point releases.

fergusstrange commented:
Hey @tachiniererin,

Thanks for the contribution. Unfortunately the point version upgrades can't be supported in OSX due to the upstream bugs in https://github.com/zonkyio/embedded-postgres-binaries/issues/21...

I'd recommend removing these from the PR and trying to stick with one clean change with the locale changes. I'll bump a minor version for this when the tests are green.

If you could also add a locale specific happy path test that would be awesome.

Thanks

tachiniererin commented:
Ok done. Reverted the version changes and added a test. Unfortunately "C" is the only locale we can always assume to work but it tests the functionality just fine.

fergusstrange commented:
Looks like there are still some failing tests here https://github.com/fergusstrange/embedded-postgres/pull/12/checks?check_run_id=761577382

Are these failing locally for you?

tachiniererin commented:
I tested it on my system with broken locale settings, so of course it passed locally :woman_facepalming:

fergusstrange commented:
Thanks @tachiniererin, release has been made at v1.1.0 https://github.com/fergusstrange/embedded-postgres/releases/tag/v1.1.0

Hope this helps you out, enjoy your weekend 👍
